app/src/main/java/com/brainiacs/seandroidapp/TeacherDashboard/ButtonAdapter.java

```diff
@@ -0,0 +1,139 @@
+    @Override
+    public Object getItem(int position) {
+        return null;
+    }
+
+    //Returns position in the
+    @Override
+    public long getItemId(int position) {
+        return position;
+    }
+
+    //Sets up buttons on grid.
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        Button btn;
+        if (convertView == null) {
+            btn = new Button(mContext);
+            btn.setLayoutParams(new GridView.LayoutParams(500, 300));
+            btn.setPadding(8, 8, 8, 8);
+        }
+            else{
+                btn = (Button) convertView;
+            }
+
+            //TODO setButtons();
+            //Sets up button text and makes them buttons
+            btn.setText(buttons[position]);
+
+            //---------------- Apply RGB Values here -------------------
+            btn.setBackgroundColor(Color.rgb(colorArray[i], colorArray[i+1], colorArray[i+2]));
+            if (i + 2 == colorArray.length - 1) {
+                i = 0;
+            }
+            else {
+                i = i + 3;
+            }
+            btn.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if(((Button) v).getText().toString().equals("Create New Class")){
+                        AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
+                        builder.setTitle("New Class Name");
+                        final EditText input = new EditText(mContext);
+                        input.setInputType(InputType.TYPE_CLASS_TEXT);
+                        builder.setView(input);
+
+                        // Set up the buttons
+                        builder.setPositiveButton("Create", new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int which) {
+                                input.getText().toString();
+                                //TODO POST new class to db
+                                Intent intent = new Intent(mContext, ClassHomeActivity.class);
+                                intent.putExtra("className", input.getText().toString());
+                                mContext.startActivity(intent);
+                            }
+                        });
+                        builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int which) {
+                                dialog.cancel();
+                            }
+                        });
+
+                        builder.show();
+                    }
+                    else {
+                        Intent intent = new Intent(mContext, ClassHomeActivity.class);
+                        intent.putExtra("className", ((Button) v).getText().toString());
```

Comment:
Make className a public static final string that can be accessed through the class.